tests/aws/services/events/test_events_inputs.py

```diff
@@ -210,3 +264,193 @@ def test_put_events_with_input_transformation_to_sqs(
+            "InputTemplate": input_template,
+        }
+        messages_match_all = put_events_with_filter_to_sqs(
+            pattern=TEST_EVENT_PATTERN,
+            entries_asserts=entries_asserts,
+            input_transformer=input_transformer_match_all,
+        )
+
+        # input transformer with keys in template missing from message
+        input_path_map_missing_key = {
+            "detail-type": "$.detail-type",
+            "timestamp": "$.time",
+            "command": "$.detail.notinmessage",
+        }
+        input_transformer_not_match_all = {
+            "InputPathsMap": input_path_map_missing_key,
+            "InputTemplate": input_template,
+        }
+        messages_not_match_all = put_events_with_filter_to_sqs(
+            pattern=TEST_EVENT_PATTERN,
+            entries_asserts=entries_asserts,
+            input_transformer=input_transformer_not_match_all,
+        )
+
+        snapshot.add_transformer(
+            [
+                snapshot.transform.key_value("MD5OfBody"),
+                snapshot.transform.key_value("ReceiptHandle"),
+            ]
+        )
+        snapshot.match("custom-variables-match-all", messages_match_all)
+        snapshot.match("custom-variables-not-match-all", messages_not_match_all)
+
+    @markers.aws.validated
+    @pytest.mark.skipif(
+        not is_v2_provider(),
+        reason="V1 provider does not support this feature",
+    )
+    def test_put_events_with_input_transformer_missing_keys(
+        self,
+        create_sqs_events_target,
+        events_create_event_bus,
+        events_put_rule,
+        aws_client_factory,
+        aws_client,
+        snapshot,
+    ):
+        _, queue_arn = create_sqs_events_target()
+
+        bus_name = f"test-bus-{short_uid()}"
+        events_create_event_bus(Name=bus_name)
+
+        rule_name = f"test-rule-{short_uid()}"
+        events_put_rule(
+            Name=rule_name,
+            EventBusName=bus_name,
+            EventPattern=json.dumps(TEST_EVENT_PATTERN),
+        )
+
+        events_client = aws_client_factory(config=Config(parameter_validation=False)).events
+        target_id = f"target-{short_uid()}"
+        input_path_map = {
+            "detail-type": "$.detail-type",
+            "timestamp": "$.time",
+            "command": "$.detail.command",
+        }
+        # input template with not defined key
+        input_template = '"Event of <detail-type> type, with not defined key <notdefinedkey>"'
+        input_transformer = {
+            "InputPathsMap": input_path_map,
+            "InputTemplate": input_template,
+        }
+
+        with pytest.raises(Exception) as exception:
+            events_client.put_targets(
+                Rule=rule_name,
+                EventBusName=bus_name,
+                Targets=[
+                    {"Id": target_id, "Arn": queue_arn, "InputTransformer": input_transformer},
+                ],
+            )
+
+        snapshot.add_transformer(
+            [
```

Comment:
nit: list not needed

Reply:
fixed in: https://github.com/localstack/localstack/pull/10789/commits/da00fe393254ac029ca9f0680166ee7242ea102f